app/sockets/service.py

```diff
@@ -101,7 +101,33 @@ def create_conversation(conversation_id, creator, participants):
                 result.append(participant)
                 
         return result
+
+    @staticmethod
+    def remove_global_identifier(email):
+        """
+        Removes the provided email from the internal mapping
+        of users to socket IDs
+        
+        :email        the email address of a user
+        """
+        
+        if email in SocketService._sockets:
+            del SocketService._sockets[email]
     
+    @staticmethod
+    def _leave_conversation(socket_id, conversation_id, participant):
```

Comment:
This should probably not be private, since the `SocketSetupService` will need to call this somewhere (remove the `_`)